shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java

```diff
@@ -240,21 +242,24 @@ public void onSystemUiVisibilityChange(int visibility) {
   private void setSystemChromeEnabledSystemUIMode(PlatformChannel.SystemUiMode systemUiMode) {
     int enabledOverlays;
 
-    if (systemUiMode == PlatformChannel.SystemUiMode.LEAN_BACK
-        && Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
+    Window window = activity.getWindow();
+    View view = window.getDecorView();
+    WindowInsetsControllerCompat windowInsetsControllerCompat =
+        new WindowInsetsControllerCompat(window, view);
+
+    if (systemUiMode == PlatformChannel.SystemUiMode.LEAN_BACK) {
       // LEAN BACK
       // Available starting at SDK 16
       // Should not show overlays, tap to reveal overlays, needs onChange callback
       // When the overlays come in on tap, the app does not receive the gesture and does not know
       // the system overlay has changed. The overlays cannot be dismissed, so adding the callback
       // support will allow users to restore the system ui and dismiss the overlays.
       // Not compatible with top/bottom overlays enabled.
-      enabledOverlays =
-          View.SYSTEM_UI_FLAG_LAYOUT_STABLE
-              | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
-              | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
-              | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
-              | View.SYSTEM_UI_FLAG_FULLSCREEN;
+      currentSystemUiMode = PlatformChannel.SystemUiMode.LEAN_BACK;
+      windowInsetsControllerCompat.setSystemBarsBehavior(
+          WindowInsetsControllerCompat.BEHAVIOR_SHOW_BARS_BY_TOUCH);
+      windowInsetsControllerCompat.hide(WindowInsetsCompat.Type.systemBars());
+      WindowCompat.setDecorFitsSystemWindows(window, false);
```

Comment:
`setDecorFitsSystemWindows(window, false)` means edge-to-edge. 

Are all the values in https://github.com/flutter/flutter/blob/61a0add2865c51bfee33939c1820709d1115c77d/packages/flutter/lib/src/services/system_chrome.dart#L162 equivalent edge-to-edge? @Piinks

Reply:
I am not sure I understand the question, are all the values in <what> equivalent to edge to edge?

Reply:
In the enum SystemUiMode

Reply:
No, only one is edge to edge, the rest are different?

Reply:
Thanks.

It sounds like only the case `PlatformChannel.SystemUiMode.EDGE_TO_EDGE` would be `setDecorFitsSystemWindows(window, false)`.  I don't know if there's a better way to test this than manually trying each mode and seeing the before and after this PR.

Reply:
That class replaces the flag [SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN](https://developer.android.com/reference/android/view/View#SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN) which is also a part of other modes

Reply:
Thanks for the videos. It does seem like all the modes that Flutter supports are edge-to-edge in Android terminology. https://developer.android.com/training/gestures/edge-to-edge - I didn't see any example where the opposite to that definition is true.

Reply:
It seems like to implement edge to edge mode like that shown in the guide, you have to do some further style customization besides using this method (like to get the transparent bars).